docs/layers.md

```diff
@@ -2108,6 +2108,90 @@ Transit lines may have their colours mapped onto one of these CSS colours. The i
 * `yellow`
 * `yellowgreen`
 
+## Traffic Flow
+
+This **optional** layer is meant to be updated every few minutes, due to the highly dynamic nature of the data. Consideration should be given to rendering the features from this layer linked to the **roads** layer when linear referencing is available, or as an overlay
+
+* Layer name: `traffic_flow`
+* Geometry types: `line`
+
+#### `common` properties:
+
+* `id`: Unique traffic event ID. Can be referenced when checking for updated traffic information for specified event
+* `kind`: the **severity** of the flow information as seen by traffic provider to indicate the traffic "color"
+* `min_zoom`: this value is derived from combination of `road_kind`, `kind`, and `kind_detail` values. A suggestion for which zoom to draw a feature. The value is a float
+
+#### Possible `kind` values for traffic flow:
+
+* `unknown`: traffic status unknown
+* `free`: Free flowing, not disturbed, traffic
+* `minor`: Minor traffic
+* `slow`: Slowly moving traffic
+* `queuing`: The traffic is in queues but still moves slowly
+* `stationary`: Stationary traffic, congestion
+* `none`: No traffic flow due to blockage or closure
+
+#### `common-optional` properties:
+
+* `source`: provider of traffic information
+* `congestion`: the level of traffic flow, with 0.0 representing completely free flowing traffic to 1.0 completely congested traffic
+* `speed`: speed in km/h
+* `drives_on_left`: set to `true` when the country drives on the left, e.g. In the U.K
+
+#### `optional` properties:
+
+The following properties should only be included when **roads** layer features doesn't include `linear_ref_id`: identifier link to a linear references system, eg [SharedStreet](https://github.com/sharedstreets/sharedstreets-ref-system/blob/master/OSMLR.md). In that case the traffic data should be delivered without geometry and run-time linked with the **roads** layer. But baring that, the following `optional` properties enable sizing and layering of the traffic overlay with respect to the roads network:
+
+* `linear_ref_id`: identifier link to a linear references system, eg [SharedStreet](https://github.com/sharedstreets/sharedstreets-ref-system/blob/master/OSMLR.md), or..
+* `road_kind`: [kind of the road](https://github.com/tilezen/vector-datasource/blob/master/docs/layers.md#road-properties-common)
+* `is_bridge`: set to `true` when the linear is a bridge
+* `is_tunnel`: set to `true` when the linear is a tunnel
+* `is_link`: set to true when the linear is a slip-road
```

Comment:
It can be helpful for people writing styles to have [a `sort_rank` property](https://github.com/tilezen/vector-datasource/blob/935a6ee920464379275ace3d29c3ef17ec94a255/docs/layers.md#feature-ordering) to guide the draw order of features. Of course, any style can override or ignore this, but it often greatly simplifies the work of writing styles if a default is provided. Tilezen includes one in the [**roads** layer common properties](https://github.com/tilezen/vector-datasource/blob/935a6ee920464379275ace3d29c3ef17ec94a255/docs/layers.md#road-properties-common), and I think it would be helpful to do the same if the **traffic flow** layer includes geometry.

Reply:
With regards to styling the traffic flow - one will be provided with both the raw traffic congestion levels (in the `congestion`) field and the _suggested_ traffic congestion levels from the `kind` values. 

Therefore, the ordering can be made from the order of the `kind` values where `unknown` would be drawn first and `none` last - as it is the most severe. Also, if one wishes to have more granular control over the ordering, the style may also be made from the congestion level where 0.0 be drawn first and 1.0 last. Up to the stylist in this case. What do you think?

Reply:
I agree that there's enough information for the stylist to do anything that they want. My point was that it's very helpful to the stylist to have that pre-encoded in a `sort_rank` property that's compatible with all the other layers. That way, it's easier to get started with a style that looks right, and we don't have to include a bunch of "polyfill" code to get a sensible-looking map.

Reply:
Okay, this makes sense if all other layers can also use this feature. Would you agree that we can make this one optional for the traffic layer?

Reply:
Great, thanks! Yes, it can be optional.

Would it also be useful for linear features in the **traffic_incidents** layer too, or are they intended to be rendered as points or above road labels?